controllers/volumereplicationgroup_controller.go

```diff
@@ -315,6 +347,12 @@ type VRGInstance struct {
 	vrcUpdated    bool
 }
 
+type timerInstance struct {
+	timer   prometheus.Timer
+	state   ramendrv1alpha1.State
```

Comment:
I hadn't considered this, but it's indeed possible to remove `desiredState` and replace by a null check on the `timer` variable. Done on 15e9fc4fff5c84b2e22a57f17a79fe5ede9f593b

Reply:
I was referring to `timerInstance.state` being not needed.  Its meaning seems to be the primary or secondary state when a timer is started.   Desired state seems to be the state to stop the timer on.  I was thinking start timer state need not be preserved since timer is started whenever reconcile is called on a new vrg.  But as mentioned in https://github.com/RamenDR/ramen/pull/186#discussion_r733330646 the "new" vrg assumption may be false.